subsys/fs/zffs/src/zffs_misc.c

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2018 Findlay Feng
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "zffs/misc.h"
+#include "zffs/dir.h"
+#include "zffs/file.h"
+#include "zffs/object.h"
+#include "zffs/path.h"
+#include "zffs/tree.h"
+
+void zffs_misc_lock(struct zffs_data *zffs)
+{
+	k_mutex_lock(&zffs->lock, K_FOREVER);
+}
+
+void zffs_misc_unlock(struct zffs_data *zffs)
+{
+	k_mutex_unlock(&zffs->lock);
+}
+
+u32_t zffs_misc_get_id(struct zffs_data *zffs)
+{
+	int lock = k_mutex_lock(&zffs->lock, K_NO_WAIT);
+	int id;
+
+	if (zffs->next_id == ZFFS_ROOT_ID) {
```

Comment:
So getting the lock here isn't a condition to actually temper zffs content? Worth verifying this.

Reply:
Now I think this lock can be deleted.
When adding this function, it was considered that it might be called outside the zffs environment.
For now, this is unlikely.